src/Collapse.php

```diff
@@ -4,195 +4,418 @@
 
-    protected function toggleComponent(): string
+    /**
+     * Adds a sets of attributes.
+     *
+     * @param array $attributes Attribute values indexed by attribute names. e.g. `['id' => 'my-id']`.
+     *
+     * @return self A new instance with the specified attributes added.
+     *
+     * Example usage:
+     * ```php
+     * $collapse->addAttributes(['data-id' => '123']);
+     * ```
+     */
+    public function addAttributes(array $attributes): self
     {
-        return 'collapse';
+        $new = clone $this;
+        $new->attributes = [...$this->attributes, ...$attributes];
+
+        return $new;
     }
 
-    public function withOptions(array $options): self
+    /**
+     * Adds one or more CSS classes to the existing classes.
+     *
+     * Multiple classes can be added by passing them as separate arguments. `null` values are filtered out
+     * automatically.
+     *
+     * @param BackedEnum|string|null ...$class One or more CSS class names to add. Pass `null` to skip adding a class.
+     *
+     * @return self A new instance with the specified CSS classes added to existing ones.
+     *
+     * @link https://html.spec.whatwg.org/#classes
+     *
+     * Example usage:
+     * ```php
+     * $collapse->addClass('custom-class', null, 'another-class', BackGroundColor::PRIMARY);
+     * ```
+     */
+    public function addClass(BackedEnum|string|null ...$class): self
     {
         $new = clone $this;
-        $new->options = $options;
+        $new->cssClasses = [...$this->cssClasses, ...$class];
 
         return $new;
     }
 
-    public function withBodyOptions(array $options): self
+    /**
+     * Adds a CSS style.
+     *
+     * @param array|string $style The CSS style. If an array, the values will be separated by a space. If a string, it
+     * will be added as is. For example, `color: red`. If the value is an array, the values will be separated by a
+     * space. e.g., `['color' => 'red', 'font-weight' => 'bold']` will be rendered as `color: red; font-weight: bold;`.
+     * @param bool $overwrite Whether to overwrite existing styles with the same name. If `false`, the new value will be
+     * appended to the existing one.
+     *
+     * @return self A new instance with the specified CSS style value added.
+     *
+     * Example usage:
+     * ```php
+     * $collapse->addCssStyle('color: red');
+     *
+     * // or
+     * $collapse->addCssStyle(['color' => 'red', 'font-weight' => 'bold']);
+     * ```
+     */
+    public function addCssStyle(array|string $style, bool $overwrite = true): self
     {
         $new = clone $this;
-        $new->bodyOptions = $options;
-
-        if (!array_key_exists('tag', $options)) {
-            $new->bodyOptions['tag'] = 'div';
-        }
+        Html::addCssStyle($new->attributes, $style, $overwrite);
 
         return $new;
     }
 
-    public function withContent(string|Stringable $content): self
+    /**
+     * Adds a sets attribute value.
```

Comment:
```suggestion
     * Sets attriute value.
```

Reply:
But the method adds an existing attribute to the general attributes, and it is the same phpdoc in `yiisoft/html`.